src/main/java/algorithms/curated170/medium/campusbikes/CampusBikesBucketSort.java

```diff
@@ -0,0 +1,93 @@
+package algorithms.curated170.medium;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class CampusBikesBucketSort {
+
+      int w, b;
+      List<int[]>[] buckets = new List[2001];
+
+      public int[] assignBikes(int[][] workers, int[][] bikes) {
+
+            w = workers.length;
+            b = bikes.length;
+
+            assignToBuckets(workers, bikes);
+
+            return assignFromBuckets();
+      }
+
+      private void assignToBuckets(int[][] workers, int[][] bikes) {
+            for (int i = 0; i < w; i++) {
+                  for (int j = 0; j < b; j++) {
+                        int dist = dist(workers[i], bikes[j]);
+                        if (buckets[dist] == null) {
+                              buckets[dist] = new ArrayList<>();
+                        }
+                        
+                        buckets[dist].add(new int[] { i, j });
+                  }
+            }
+      }
+
+      private int[] assignFromBuckets() {
+            int[] wo = new int[w];
+            boolean[] bikeAssigned = new boolean[w];
+            boolean[] workerAssigned = new boolean[b];
```

Comment:
If we fix these:
```
            boolean[] bikeAssigned = new boolean[b];
            boolean[] workerAssigned = new boolean[w];
```
![image](https://user-images.githubusercontent.com/60903744/118286888-996fa380-b4db-11eb-8f4f-073e29a2cdae.png)

Reply:
I have fixed it, actually at the same moment without seeing this comment.

Reply:
How about the Priority Queue solution?

Reply:
It is a little bit slow.
But, maybe it is normal for PQ.
![image](https://user-images.githubusercontent.com/60903744/118288412-15b6b680-b4dd-11eb-99ea-acc6fdfb91df.png)